src/utils.rs

```diff
@@ -42,33 +34,146 @@ where
     }
 }
 
-/// Spawn a task and abort process if it panics or results in error.
-pub fn spawn_or_abort<F, T>(future: F) -> JoinHandle<T>
+pub fn spawn_monitored_with_backoff<S, F, T>(
```

Comment:
This handles the spawning and respawning of task futures, while waiting after every failure